packages/cli/src/cmds/preview/preview.ts

```diff
@@ -822,4 +849,35 @@ export default class Preview extends YargsCommand {
     }
     this.backgroundTasks = [];
   }
+
+  private async handleDependences(hasBackend: boolean): Promise<[FuncToolChecker, DotnetChecker]> {
+    const cliAdapter = new CLIAdapter(cliEnvCheckerTelemetry, hasBackend);
+    const nodeChecker = new AzureNodeChecker(
+      cliAdapter,
+      cliEnvCheckerLogger,
+      cliEnvCheckerTelemetry
+    );
+    const dotnetChecker = new DotnetChecker(
+      cliAdapter,
+      cliEnvCheckerLogger,
+      cliEnvCheckerTelemetry
+    );
+    const funcChecker = new FuncToolChecker(
+      cliAdapter,
+      cliEnvCheckerLogger,
+      cliEnvCheckerTelemetry
+    );
+    const depsChecker = new DepsChecker(cliEnvCheckerLogger, cliAdapter, [
+      nodeChecker,
+      dotnetChecker,
+      funcChecker,
+    ]);
+
+    const shouldContinue = await depsChecker.resolve();
+    if (!shouldContinue) {
+      throw new Error("preview stopped.");
```

Comment:
Define an error in `errors.ts` and return that error instead of throwing an error directly.

Reply:
added